### DISCUSSION TITLE
Keyboard backlight modes not working

### DISCUSSION THREAD

**Discussion by ZEREF112** [seerge/g-helper#1255] [Q&A]
Hello friends, I have this problem and I need some help. Keyboard backlight settings not working. Whatever you do, the lighting method does not change. My  Laptop is TUF Dash F15

**Comment by seerge** [comment]
@ZEREF112 hello, i don't have telepathic skills do detect what's wrong in your case. 

Post a log here as explained in Issues. Also did it work before, or it never worked for you ?

**Reply by ZEREF112** [reply]
The keyboard backlight modes never worked for me. Where can I find the log file so I can send it to you?

**Reply by seerge** [reply]
At ``%appdata%\ghelper\``

**Reply by ZEREF112** [reply]
[log.txt](https://github.com/seerge/g-helper/files/12525366/log.txt)

**Reply by seerge** [reply]
Well, i see that your laptop is rejecting an RGB setting

```
9/5/2023 2:36:29 PM: TUF RGB = B4-00-29-94-00-00-00-00-00-00-00-00 : 0
9/5/2023 2:36:31 PM: TUF RGB = B4-00-29-94-00-00-00-00-00-00-00-00 : 0
```

It's the same command for all TUF models. I'm not aware about anything else.

Does OpenRGB or any other open-source software able to control your backlight ?

**Reply by ZEREF112** [reply]
I haven't tried any other software I'll try with OpenRGB to see if it works

**Reply by ZEREF112** [reply]
I just tried openRGB and it didn't work the keyboard keeps blinking I just want to make it Static

**Reply by seerge** [reply]
If it's blinking most probably your keyboard by some reason thinks that laptop is sleeping. Does turning off "Sleep" checkbox under Extra settings made any difference ?

I have made a small tweak in a command , try this build (most probably it will behave same)

**[GHelper.zip](https://github.com/seerge/g-helper/files/12526310/GHelper.zip)**

**Reply by ZEREF112** [reply]
I tried it now. The keyboard's still blinking.

**Reply by seerge** [reply]
@ZEREF112 then try to do a [hardware reset ](https://github.com/seerge/g-helper#how-do-i-do-a-hardware-reset-on-a-laptop) it will probably set keyboard into default state as well

**Reply by seerge** [reply]
@ZEREF112 check if this build makes any difference ? 
[GHelper.zip](https://github.com/seerge/g-helper/files/12536811/GHelper.zip)

**Reply by ZEREF112** [reply]
It worked. I can now switch between modes. Thank you

**Reply by seerge** [reply]
Bingo
